hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsConfiguration.kt

```diff
@@ -27,28 +29,23 @@ class HmppsSqsConfiguration {
   @ConditionalOnMissingBean
   fun hmppsQueueFactory(applicationContext: ConfigurableApplicationContext) = HmppsQueueFactory(applicationContext, SqsClientFactory())
 
-  @Bean
-  @ConditionalOnMissingBean
-  fun hmppsTopicService(hmppsTopicFactory: HmppsTopicFactory, hmppsSqsProperties: HmppsSqsProperties) =
-    HmppsTopicService(hmppsTopicFactory, hmppsSqsProperties)
-
   @Bean
   @ConditionalOnMissingBean
   fun hmppsQueueService(
     telemetryClient: TelemetryClient?,
-    hmppsTopicService: HmppsTopicService,
+    hmppsTopicFactory: HmppsTopicFactory,
     hmppsQueueFactory: HmppsQueueFactory,
     hmppsSqsProperties: HmppsSqsProperties,
-  ) = HmppsQueueService(telemetryClient, hmppsTopicService, hmppsQueueFactory, hmppsSqsProperties)
+  ) = HmppsQueueService(telemetryClient, hmppsTopicFactory, hmppsQueueFactory, hmppsSqsProperties)
 
   @Bean
   @ConditionalOnMissingBean
-  @ConditionalOnExpression("'\${hmpps.sqs.reactiveApi:false}'.equals('false')")
+  @ConditionalOnWebApplication(type = SERVLET)
```

Comment:
this looks a nicer way of deciding which resource to use - will be decided by spring instead